src/core/domain/entities/pedido.ts

```diff
@@ -0,0 +1,73 @@
+import { AssertionConcern } from "../base/assertionConcern";
+import { Entity } from "../base/entity.interface";
+import { Produto } from "./produto";
+
+export enum StatusPedidoEnum {
+    Recebido = "recebido",
+    Em_preparacao = "em_preparacao",
+    Pronto = "pronto",
+    Finalizado = "finalizado",
+}
+
+export type StatusPedido = `${StatusPedidoEnum}`;
+
+export interface Item {
+    produto: Produto;
+    quantidade: number;
+}
+
+export class Pedido implements Entity {
+    id: string;
+    status: StatusPedido;
+    valorTotal: number; // float
+    // cliente?: Cliente; // user
+    cliente?: string;
+    itens: Item[];
+    observacoes?: string;
+
+    constructor({
+        id,
+        status,
+        valorTotal,
+        cliente,
+        itens,
+        observacoes,
+    }: {
+        id: string;
+        status: StatusPedido;
+        valorTotal: number;
+        // cliente?: Cliente;
+        cliente?: string;
+        itens: Item[];
+        observacoes?: string;
```

Comment:
@mitestainer 
- só uma dúvida aqui o `valorTotal` está sendo definido direto com oq vem do body?
- O ideal aqui é que esse valor final seja calculado aqui na entidade com base nos itens, para evitar de mandarem qualquer coisa no body


ps: Ainda não testei o código pq acho que não está completo

Reply:
Concordo, só não to conseguindo fazer 😅. Dá uma mão?